types.go

```diff
@@ -21,6 +22,9 @@ const (
 )
 
 func (e EvaluationType) String() string {
+	if e < 0 || int(e) >= len(evaluationStrings) {
+		return fmt.Sprintf("EvaluationType(%d)", e)
```

Comment:
Sorry,  that wasn't my question. I am asking about the fmt call. e is a
Stringer. I think to print the actual integer value you need to cast to
int. Maybe that is overwritten by %d
On Jan 21, 2015 6:15 PM, "Vladimír Chalupecký" <notifications@github.com>
wrote:

> In types.go
> <https://github.com/gonum/optimize/pull/45#discussion_r23351472>:
>
> > @@ -21,6 +22,9 @@ const (
> >  )
> >
> >  func (e EvaluationType) String() string {
> > +	if e < 0 || int(e) >= len(evaluationStrings) {
> > +		return fmt.Sprintf("EvaluationType(%d)", e)
>
> Yes, it works, because 0 is an untyped integer constant that can be
> compared or assigned to any compatible integer variable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gonum/optimize/pull/45/files#r23351472>.
>